**COMPARE** **WITH** **PRETRAINED** **MODEL**

In [4]:
!pip install transformers
import transformers
!pip install transformers[sentencepiece]
from transformers import pipeline


In [5]:
summarizer = pipeline("summarization",model="facebook/bart-large-cnn")
summarizer("""A Standard Change is created from preapproved Standard change templates that have satisfied specific criteria (see Appendix A) and been added to the Standard Change Template Catalog in the Enterprise Ticketing System: the change is repeatable, frequently implemented, is considered low risk and low impact according to the SIA, and has a proven history of success (completed the Normal change lifecycle process at least 3-5 times with no issues). Standard changes that are approved by Division Directors and CAB Co-Chairs are added to the catalog and considered pre-authorized, following a shorter ChM lifecycle outside of the CAB approval process (subject to dual authorization). CRs can request new Standard change templates or use an existing template from the catalog to create a new Standard change request.A Normal Change is one that meets the defined lead time for testing and validation and is assigned a SIA level of no, low, medium, or high. A Normal change is an RFC that is not a Standard, Expedited or Emergency change, and is subject to the full ChM review process, including review and authorization by the CAB.An Expedited Change does not meet the lead time requirement for a Normal change, but is not an Emergency Change. Service is at risk, although service might not be down, and the RFC needs to be authorized because of a client request that has been validated by SME/ technical expert or a Director, who has determined that that the change needs to go in without waiting for the recommended lead-time. The same ‘Normal’ Change request information is provided in the Enterprise Ticketing System to implement the change, including the reason for expediting the RFC (SIA, back-out plans, scheduled time and downtime required), but lead times are much shorter. Authorization by a CAB Co-Chair is required and Expedited Changes are subject to retroactive review by CAB. Emergency Change is one that must be implemented as soon as possible to correct, or prevent, a high priority incident, or that must be introduced as soon as possible due to likely negative service impacts or situations where the impact to a service is imminent if action is not taken. These changes do not follow the complete lifecycle of a Normal change due to the speed with which they must be implemented and authorized. All emergency changes are authorized by E-CAB and documented and entered into the Enterprise Ticketing System prior to implementation, or as soon as possible after the change has been implemented depending on the nature of the emergency.Emergency changes are subject to a Post Implementation Review (PIR) process by CAB.
Write an IT Audit control what to review for the above text
Summary:
"""
)

/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_token.py:88: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


config.json:   0%|          | 0.00/1.58k [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/1.63G [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/363 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/899k [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

[{'summary_text': 'A Normal Change is one that meets the defined lead time for testing and validation and is assigned a SIA level of no, low, medium, or high. An Expedited Change does not meet the lead time requirement for a Normal change, but is not an Emergency Change. Emergency changes are subject to a Post Implementation Review (PIR) process by CAB.'}]

**ROUGE** **EVALUATION**

In [6]:
pip install rouge_score

  Using cached rouge_score-0.1.2-py3-none-any.whl


In [7]:
from rouge_score import rouge_scorer

def evaluate_rouge(hypotheses, references):
    scorer = rouge_scorer.RougeScorer(['rouge1', 'rouge2', 'rougeL'], use_stemmer=True)
    scores = scorer.score(hypotheses, references)
    return scores


hypotheses = "Review whether the company prioritizes and categorizes changes by whether they are approved by the CAB, subject to the approval of a formal process, and if they include the submission of change requests, approval of changes, and testing before implementation in the production environment."
references = "Application changes are appropriately tested and approved before being moved into the production environment"

rouge_scores = evaluate_rouge(hypotheses, references)
print(rouge_scores)

{'rouge1': Score(precision=0.6428571428571429, recall=0.20454545454545456, fmeasure=0.31034482758620696), 'rouge2': Score(precision=0.15384615384615385, recall=0.046511627906976744, fmeasure=0.07142857142857142), 'rougeL': Score(precision=0.5714285714285714, recall=0.18181818181818182, fmeasure=0.27586206896551724)}


{'rouge1': Score(precision=0.6428571428571429, recall=0.20454545454545456, fmeasure=0.31034482758620696), 'rouge2': Score(precision=0.15384615384615385, recall=0.046511627906976744, fmeasure=0.07142857142857142), 'rougeL': Score(precision=0.5714285714285714, recall=0.18181818181818182, fmeasure=0.27586206896551724)}
**κείμενο με έντονη γραφή**

In [8]:
from rouge_score import rouge_scorer

def evaluate_rouge(hypotheses, references):
    scorer = rouge_scorer.RougeScorer(['rouge1', 'rouge2', 'rougeL'], use_stemmer=True)
    scores = scorer.score(hypotheses, references)
    return scores


hypotheses = "Review whether the company prioritizes and categorizes changes according to the implementation of a formal process and verify that it includes the submission of change requests, approval of changes, and testing before implementation in the production environment. Ensure that all changes are subject to the same standards and that they comply with the same requirements as those of the release of new releases, security policies, and procedures."
references = "Application changes are appropriately tested and approved before being moved into the production environment"

rouge_scores = evaluate_rouge(hypotheses, references)
print(rouge_scores)

{'rouge1': Score(precision=0.6428571428571429, recall=0.13432835820895522, fmeasure=0.2222222222222222), 'rouge2': Score(precision=0.23076923076923078, recall=0.045454545454545456, fmeasure=0.0759493670886076), 'rougeL': Score(precision=0.5, recall=0.1044776119402985, fmeasure=0.1728395061728395)}


{'rouge1': Score(precision=0.6428571428571429, recall=0.13432835820895522, fmeasure=0.2222222222222222), 'rouge2': Score(precision=0.23076923076923078, recall=0.045454545454545456, fmeasure=0.0759493670886076), 'rougeL': Score(precision=0.5, recall=0.1044776119402985, fmeasure=0.1728395061728395)}
